protected/humhub/modules/content/services/ContentSearchService.php

```diff
@@ -15,11 +15,21 @@
 
 class ContentSearchService
 {
-    public Content $content;
+    public ?Content $content;
+    public ?int $contentId;
 
-    public function __construct(Content $content)
+    /**
+     * @param Content|int $content
+     */
+    public function __construct($content)
```

Comment:
@yurabakhtin I would prefer when the ContentSearch service always reliably has a non-null $content record.

What do you think of the following solution, we create a new static method `ContentSearchService::deleteContentById($id, $asActiveJob)`  
and also use this method in `ContentSearchService::delete($asActiveJob)`.

Reply:
@luke- Done in the commit https://github.com/humhub/humhub/pull/7187/commits/5b8741fd3e237b7b3b6977f518f4231c2c764b53.

What do you think if we will implement new static method `instance()` for the service like this? :

```php
    public static function instance(Content $content): self
    {
        return new self($content);
    }
```

so then we will update all codes from:

- `(new ContentSearchService($content))->methodName()`

to:

- `ContentSearchService::instance($content)->methodName()`

I think it looks better without the brackets `()` around.

Reply:
@yurabakhtin With a method `className::instance()` (getInstance) I always somehow think it is a singleton pattern. https://en.wikipedia.org/wiki/Singleton_pattern

---

Since this affects many other service classes too, we should try to find here a solution/style we use in the whole app. I've createda seperate issue for it here: https://github.com/humhub/humhub-internal/issues/347